### ISSUE TITLE
Structr won't build using java 8 due to missing javadoc items

### ISSUE THREAD

**Issue by raginggoblin** [structr/structr#368]
Apparently javadoc plugin is somewhat strict about missing stuff with java 8:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9.1:jar (attach-javadocs) on project structr-core: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - Picked up JAVA_TOOL_OPTIONS: -javaagent:/usr/share/java/jayatanaag.jar
[ERROR] ~/structr/structr-core/src/main/java/org/structr/agent/Agent.java:260: warning: no description for @param

**Comment by amorgner** [comment]
This is due to JDK 8 requiring Javadoc to comply with doclint, see this good blog post (with an interesting discussion): http://blog.joda.org/2014/02/turning-off-doclint-in-jdk-8-javadoc.html

We could either just disable doclint, or fix the Javadoc, which turns out to not being trivial, but one commenter suggests the following pom.xml settings to make it build with JDK 8 while still being compatible with JDK 7:

    [...]
        <profile>
            <id>java8-disable-strict-javadoc</id>
            <activation>
                <jdk>[1.8,)</jdk>
            </activation>
            <properties>
                <javadoc.doclint.none>-Xdoclint:none</javadoc.doclint.none>
            </properties>
        </profile>
    [...]

and

    [...]
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <version>2.10.1</version>
                <configuration>
                    <additionalparam>${javadoc.doclint.none}</additionalparam>
                    <quiet>true</quiet>
                </configuration>
                <reportSets>
                    <reportSet>
                        <id>html</id>
                        <reports>
                            <report>javadoc</report>
                        </reports>
                    </reportSet>
                </reportSets>
            </plugin>
    [...]

We'll make some tests based on this  to see how we can make Structr build successfully with JDK 7 and 8.

**Comment by amorgner** [comment]
It's even (much) simpler, the following global setting is already sufficient, I just picked the wrong settings from the blog post's comment first.

    <properties>
    [...]
        <additionalparam>-Xdoclint:none</additionalparam>
    [...]
    </properties>

One other thing I had to change was the timezone setting for tests.  JDK 8 reads the timezone information earlier than before, see http://stackoverflow.com/questions/23466218/setting-timezone-for-maven-unit-tests-on-java-8

With commit 8f314dfc2bba7ecd0ace8b3b27c27c7d33939521, Structr builds fine on JDK 8 and 7, and all tests pass.
